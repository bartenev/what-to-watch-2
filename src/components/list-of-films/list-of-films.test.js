import React from "react";
import renderer from "react-test-renderer";
import ListOfFilms from "./list-of-films";

const films = [
  {
    title: `Fantastic Beasts`,
    picture: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    title: `Bohemian Rhapsody`,
    picture: `img/bohemian-rhapsody.jpg`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,

  },
  {
    title: `Macbeth`,
    picture: `img/macbeth.jpg`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,

  },
  {
    title: `Aviator`,
    picture: `img/aviator.jpg`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,

  },
  {
    title: `We need to talk about Kevin`,
    picture: `img/we-need-to-talk-about-kevin.jpg`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];

it(`ListOfFilms is rendered correctly`, () => {
  const tree = renderer.create((
    <ListOfFilms
      films={films}
    />
  ), {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
