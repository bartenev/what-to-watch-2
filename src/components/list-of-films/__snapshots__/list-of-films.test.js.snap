// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfFilms is rendered correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Action
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Mystery
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          className="player__video"
          muted={true}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Pulp Fiction
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          className="player__video"
          muted={true}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Macbeth
        </a>
      </h3>
    </article>
  </div>
  
</section>
`;
